examples/example-hono-api/src/auth0.ts

```diff
@@ -0,0 +1,175 @@
+        auth0_domain: auth0Env.AUTH0_DOMAIN,
+        auth0_audience: auth0Env.AUTH0_AUDIENCE,
+      };
+      if (!auth0_domain || auth0_domain.length === 0) {
+        throw new Error('JWT auth middleware requires options "auth0_domain"');
+      }
+      if (!auth0_audience || auth0_audience.length === 0) {
+        throw new Error('JWT auth middleware requires options "auth0_audience"');
+      }
+
+      apiClient = new ApiClient({
+        domain: auth0_domain,
+        audience: auth0_audience,
+      });
+    }
+
+    const credentials = ctx.req.raw.headers.get('Authorization');
+    if (!credentials) {
+      const errDescription = 'No Authorization header included in request';
+      throw new HTTPException(401, {
+        message: errDescription,
+        res: unauthorizedResponse({
+          ctx,
+          error: 'invalid_request',
+          errDescription,
+        }),
+      });
+    }
+
+    const parts = credentials.split(/\s+/);
+    if (parts.length !== 2) {
+      const errDescription = 'Invalid Authorization header structure';
+      throw new HTTPException(401, {
+        message: errDescription,
+        res: unauthorizedResponse({
+          ctx,
+          error: 'invalid_request',
+          errDescription,
+        }),
+      });
+    }
+
+    if (parts[0] !== 'Bearer') {
+      const errDescription =
+        'Invalid authorization header (only Bearer tokens are supported)';
+      throw new HTTPException(401, {
+        message: errDescription,
+        res: unauthorizedResponse({
+          ctx,
+          error: 'invalid_request',
+          errDescription: errDescription,
+        }),
+      });
+    }
+
+    const accessToken = parts[1];
+    if (!accessToken || accessToken.length === 0) {
+      const errDescription = 'No token included in request';
+      throw new HTTPException(401, {
+        message: errDescription,
+        res: unauthorizedResponse({
+          ctx,
+          error: 'invalid_request',
+          errDescription,
+        }),
+      });
+    }
+
+    try {
+      const jwtPayload = await apiClient.verifyAccessToken({ accessToken });
+
+      ctx.set('jwtPayload', jwtPayload);
+
+      await next();
+    } catch (cause) {
+      console.trace(cause);
+
+      throw new HTTPException(401, {
+        message: 'Unauthorized',
+        res: unauthorizedResponse({
+          ctx,
+          error: 'invalid_token',
+          statusText: 'Unauthorized',
+          errDescription: 'Token verification failure',
+        }),
+        cause,
+      });
+    }
+
+  };
+};
+
+function unauthorizedResponse(opts: {
+  ctx: Context;
+  error: string;
+  errDescription: string;
+  statusText?: string;
+}) {
+  return new Response('Unauthorized', {
+    status: 401,
+    statusText: opts.statusText,
+    headers: {
+      'WWW-Authenticate': `Bearer realm="${opts.ctx.req.url}",error="${opts.error}",error_description="${opts.errDescription}"`,
+    },
+  });
```

Comment:
Any reason we arent returning it in the body? I know the `WWW-Authenticate` headers is what we need for spec compliance, but users typically prefer reading the body.

In the sample apps, I have always omitted the `WWW-Authenticate`, and only return the body.
In the SDKs, I ensured we have both (see [here](https://github.com/auth0/auth0-fastify/blob/early-access/packages/auth0-fastify-api/src/index.ts#L68-L78)).

This mostly to keep the sample apps easy to parse.